pkg/syncer/spec/mutators/deployment.go

```diff
@@ -64,9 +77,34 @@ func (dm *DeploymentMutator) Mutate(downstreamObj *unstructured.Unstructured) er
 	// Alias for the template.spec to improve readability.
 	templateSpec := &deployment.Spec.Template.Spec
 
+	desiredServiceAccountName := "default"
+	if templateSpec.ServiceAccountName != "" && templateSpec.ServiceAccountName != "default" {
+		desiredServiceAccountName = templateSpec.ServiceAccountName
+	}
+
+	secretList, err := dm.secretLister.ByIndex(ByWorkspaceIndexName, upstreamLogicalName.String())
+	if err != nil {
+		return fmt.Errorf("error listing secrets for cluster %s: %w", deployment.ClusterName, err)
+	}
+
+	desiredSecretName := ""
+	for _, secretObj := range secretList {
+		secret := secretObj.(*unstructured.Unstructured)
+		// Find the SA token that matches the service account name.
+		if val, ok := secret.GetAnnotations()[corev1.ServiceAccountNameKey]; ok && val == desiredServiceAccountName {
+			desiredSecretName = PrefixSecretName(secret.GetName())
+		}
+	}
```

Comment:
I am missing namespace logic. Why is the cluster name enough to find the right secret? Can't it be from another namespace?